geo/src/algorithm/convex_hull/qhull.rs

```diff
@@ -98,21 +99,36 @@ fn hull_set<T>(
             p_orth.x * p_diff.x + p_orth.y * p_diff.y
         })
         .enumerate()
-        .max_by(|(_, a), (_, b)| a.partial_cmp(b).unwrap())
-        .unwrap()
-        .0;
+        .try_fold(
+            (0, T::zero()),
+            |(farthest_idx, farthest_dist), (curr_idx, curr_dist)| {
+                // Ensure `distance` is not NaN
+                // TODO: we need a better check
+                if curr_dist != curr_dist {
+                    return Err(());
+                }
+
+                Ok(if curr_dist > farthest_dist {
+                    (curr_idx, curr_dist)
+                } else {
+                    (farthest_idx, farthest_dist)
+                })
+            },
+        )?;
 
-    // move Coordinate at furthest_point from set into hull
-    let furthest_point = swap_remove_to_first(&mut set, furthest_idx);
+    // move Coordinate at farthest_point from set into hull
```

Comment:
<img width="704" alt="Screen Shot 2021-01-01 at 12 14 01 PM" src="https://user-images.githubusercontent.com/217057/103445575-df158900-4c2a-11eb-9e4c-8d4a6c09faf2.png">

🧐 Ah yes...

Someone needs to build a `strunk-and-white` plugin for clippy.